metadata/metadata.go

```diff
@@ -289,7 +303,15 @@ func (c *Client) retry(ctx context.Context, cfg requestConfig) (string, error) {
 			continue
 		}
 
-		return resp, nil
+		defer resp.Body.Close()
+		md, err := io.ReadAll(resp.Body)
+		if err != nil {
+			logger.Errorf("Failed to read metadata server response bytes: %+v", err)
```

Comment:
Since we are here, can you turn this Errorf into a Debugf? And compute a summary to display on the last line?

Reply:
done